modules/lang-painless/src/main/java/org/elasticsearch/painless/action/PainlessExecuteAction.java

```diff
@@ -361,16 +364,20 @@ static Request parse(XContentParser parser) throws IOException {
             this.context = scriptContextName != null ? fromScriptContextName(scriptContextName) : PainlessTestScript.CONTEXT;
             if (setup != null) {
                 this.contextSetup = setup;
-                if (contextSetup.getClusterAlias() == null) {
-                    index(contextSetup.getIndex());
-                } else {
-                    index(contextSetup.getClusterAlias() + RemoteClusterAware.REMOTE_CLUSTER_INDEX_SEPARATOR + contextSetup.getIndex());
-                }
+                setIndex(false);
             } else {
                 contextSetup = null;
             }
         }
 
+        private void setIndex(boolean local) {
+            if (local || contextSetup.getClusterAlias() == null) {
+                index(contextSetup.getIndex());
```

Comment:
Can we assert here that either `contextSetup.getClusterAlias()` is either `null` or `_origin`?